examples/multiple_viewer_widget.py

```diff
@@ -0,0 +1,448 @@
+"""
+This is an example on how to have more than one viewer in the same napari window.
+Additional viewers state will be synchronized with the main viewer.
+Switch to 3d viewer will only impact main viewer.
+
+This example also contain option to enable cross that will be moved to the mouse position.
+"""
+
+from copy import deepcopy
+
+import numpy as np
+from qtpy.QtCore import Qt
+from qtpy.QtWidgets import (
+    QCheckBox,
+    QDoubleSpinBox,
+    QPushButton,
+    QSplitter,
+    QTabWidget,
+    QVBoxLayout,
+    QWidget,
+)
+from superqt.utils import qthrottled
+from packaging.version import parse as parse_version
+
+import napari
+from napari.components.layerlist import Extent
+from napari.components.viewer_model import ViewerModel
+from napari.layers import Image, Labels, Layer, Vectors
+from napari.qt import QtViewer
+from napari.utils.action_manager import action_manager
+from napari.utils.events.event import WarningEmitter
+from napari.utils.notifications import show_info
+
+NAPARI_GE_4_16 = parse_version(napari.__version__) > parse_version("0.4.16")
+
+
+def copy_layer_le_4_16(layer: Layer, name: str = ""):
+    res_layer = deepcopy(layer)
+    # this deepcopy is not optimal for labels and images layers
+    if isinstance(layer, (Image, Labels)):
+        res_layer.data = layer.data
+
+    res_layer.metadata["viewer_name"] = name
+
+    res_layer.events.disconnect()
+    res_layer.events.source = res_layer
+    for emitter in res_layer.events.emitters.values():
+        emitter.disconnect()
+        emitter.source = res_layer
+    return res_layer
+
+
+def copy_layer(layer: Layer, name: str = ""):
+    if NAPARI_GE_4_16:
+        return copy_layer_le_4_16(layer, name)
```

Comment:
For an example like this, I think it's slightly more readable if this `if` statement happens inside a single `copy_layer` function. But feel free to ignore if you disagree!

Reply:
I do not understand. You mean to merge these two functions or move `if` outside the function definition.